File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.cpp
#include <vector>
#include <cmath>
#include <iostream>

class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
       int n = cost.size();
       vector<int> dp(n);
       dp[0] = cost[0];
       dp[1] = cost[1];
       for (int i =2; i<n;i++){
            dp[i] = cost[i];
            if (dp[i-1]<=dp[i-2]){
                dp[i]+= dp[i-1];
            }
            else{
                dp[i]+= dp[i-2];
            }
       }
        if (dp[n-1]<dp[n-2]){
            return dp[n-1];
        }
        else{
            return dp[n-2];
        }
    }
};

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.ts
function minCostClimbingStairs(cost: number[]): number {
    const n = cost.length
    let dp: number[] = Array(n).fill(0)
    dp[0] = cost[0]
    dp[1] = cost[1]
    for (let i =2;i<n;i++){
        dp[i] = cost[i] + Math.min(dp[i-1],dp[i-2])
    }
    return Math.min(dp[n-2],dp[n-1])
};